client/rest/src/main/java/org/elasticsearch/client/RequestOptions.java

```diff
@@ -141,6 +166,15 @@ public void setHttpAsyncResponseConsumerFactory(HttpAsyncResponseConsumerFactory
             this.httpAsyncResponseConsumerFactory =
                     Objects.requireNonNull(httpAsyncResponseConsumerFactory, "httpAsyncResponseConsumerFactory cannot be null");
         }
+
+        /**
+         * Override the client's default for
+         * {@link RestClientBuilder#setStrictDeprecationMode strict deprecation}
+         * . Null means accept the client's default
+         */
+        public void setOverrideStrictDeprecationMode(Boolean overrideStrictDeprecationMode) {
```

Comment:
Minor comment... I think this name is a bit confusing.  My first instinct was to set this `true` if I wanted to override the client's default strict mode (because I want it to be not-strict). But that's not quite how it works, since this A) overrides and B) sets to the boolean you pass.

Maybe just `setStrictDeprecationMode(Boolean)` is fine, and it's understood/documented that if you set something it overrides the default which is strict?

Reply:
That is the name I started with but I second guessed myself. I'll switch it back.